hyperspy/io_plugins/bruker.py

```diff
@@ -967,6 +953,59 @@ def add_filename_to_general(self, item):
         item['metadata']['General']['original_filename'] = \
             self.filename.split('/')[-1]
 
+def spx_reader(filename, lazy=False):
+    with open(filename, 'br') as fn:
+        xml_str = fn.read()
+    root = ET.fromstring(xml_str)
+    sp_node = root.find("./ClassInstance[@Type='TRTSpectrum']")
+    try:
+        name = str(sp_node.attrib['Name'])
+    except KeyError:
+        name = 'Undefinded'
+        _logger.info("spectra have no name. Giving it 'Undefined' name")
+    spectrum = EDXSpectrum(sp_node)
+    mode = guess_mode(spectrum.hv)
+    results_xml = sp_node.find("./ClassInstance[@Type='TRTResult']")
+    elements_xml = sp_node.find("./ClassInstance[@Type='TRTPSEElementList']")
+    hy_spec = {'data': spectrum.data,
+               'axes': [{'name': 'Energy',
+                         'size': len(spectrum.data),
+                         'offset': spectrum.offset,
+                         'scale': spectrum.scale,
+                         'units': 'keV'}],
+               'metadata':
+               # where is no way to determine what kind of instrument was used:
+               # TEM or SEM
+               {'Acquisition_instrument': {
+                 mode: {'Detector':
+                            gen_detector_node(spectrum),
+                         'beam_energy': spectrum.hv}
+               },
+                'General': {'original_filename': filename.split('/')[-1],
+                            'title': 'EDX',
+                            'date': spectrum.date,
+                             'time': spectrum.time},
+                 'Sample': {'name': name,
+                            #'elements': TODO
```

Comment:
Some left over I guess?

Reply:
yeah, forgot to remove, the code below reads elements into dictionary if the xml node is present (if selection was made in Esprit, else there is no node).